src/framework/qobj.hpp

```diff
@@ -83,23 +82,76 @@ Qobj::Qobj(const json_t &js) {
   int_t seed = -1;
   uint_t seed_shift = 0;
   JSON::get_value(seed, "seed_simulator", config);
-
-  // Parse experiments
   const json_t &circs = js["experiments"];
-  for (const auto &circ : circs) {
-    Circuit circuit(circ, config);
-    // Override random seed with fixed seed if set
-    // We shift the seed for each successive experiment
-    // So that results aren't correlated between experiments
-    if (seed >= 0) {
+  const size_t num_circs = circs.size();
+
+  // Check if parameterized qobj
+  // It should be of the form
+  // [exp0_params, exp1_params, ...]
+  // where:
+  //    expk_params = [((i, j), pars), ....]
+  //    i is the instruction index in the experiment
+  //    j is the param index in the instruction
+  //    pars = [par0, par1, ...] is a list of different parameterizations
+  using pos_t = std::pair<uint_t, uint_t>;
+  using exp_params_t = std::vector<std::pair<pos_t, std::vector<double>>>;
+  std::vector<exp_params_t> param_table;
+  JSON::get_value(param_table, "parameterizations", config);
+
+  // Validate parameterizations for number of circuis
+  if (!param_table.empty() && param_table.size() != num_circs) {
+    throw std::invalid_argument(
+        R"(Invalid parameterized qobj: "parameterizations" length does not match number of circuits.)");
+  }
+
+  // Load circuits
+  for (size_t i=0; i<num_circs; i++) {
+    // Get base circuit from qobj
+    Circuit circuit(circs[i], config);
+    if (param_table.empty() || param_table[i].empty()) {
+      // Non parameterized circuit
+      circuits.push_back(circuit);
+    } else {
+      // Load different parameterizations of the initial circuit
+      const auto circ_params = param_table[i];
+      const size_t num_params = circ_params[0].second.size();
+      const size_t num_instr = circuit.ops.size();
+      for (size_t j=0; j<num_params; j++) {
+        // Make a copy of the initial circuit
+        Circuit param_circuit = circuit;
+        for (const auto &params : circ_params) {
+          const auto instr_pos = params.first.first;
+          const auto param_pos = params.first.second;
+          // Validation
+          if (instr_pos >= num_instr) {
+            throw std::invalid_argument(R"(Invalid parameterized qobj: instruction position out of range)");
+          }
+          auto &op = param_circuit.ops[instr_pos];
+          if (param_pos >= op.params.size()) {
+            throw std::invalid_argument(R"(Invalid parameterized qobj: instruction param position out of range)");
+          }
+          if (j >= params.second.size()) {
+            throw std::invalid_argument(R"(Invalid parameterized qobj: parameterization value out of range)");
+          }
+          // Update the param
+          op.params[param_pos] = params.second[j];
```

Comment:
how does the typing work here? Do we only allow a single parameter type? Or am I missing some clever templating?

Reply:
At the moment this only works for complex or double params (basically u1, u2, u3 gates).